flytekit/bin/entrypoint.py

```diff
@@ -549,7 +549,13 @@ def fast_execute_task_cmd(additional_distribution: str, dest_dir: str, task_exec
 
     # Use the commandline to run the task execute command rather than calling it directly in python code
     # since the current runtime bytecode references the older user code, rather than the downloaded distribution.
-    p = subprocess.Popen(cmd)
+    env = os.environ.copy()
+    if all(os.path.realpath(path) != dest_dir for path in sys.path) and dest_dir is not None:
```

Comment:
The new process may have a different `sys.path` that is not connected to the current process.

I think it's okay to just check for `dest_dir is not None`.